cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -873,6 +873,65 @@ TYPED_TEST(TestRandomNumericCountKernel, RandomArrayCount) {
   }
 }
 
+//
+// Count Distinct
+//
+
+class TestCountDistinctKernel : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    only_valid = CountOptions(CountOptions::ONLY_VALID);
+    only_null = CountOptions(CountOptions::ONLY_NULL);
+    all = CountOptions(CountOptions::ALL);
+  }
+
+  const Datum& expected(int64_t value) {
+    expected_values[value] = Datum(static_cast<int64_t>(value));
+    return expected_values.at(value);
```

Comment:
Not sure I understand the rationale here. You are basically storing and returning `Datum(value)`?

Reply:
Indeed, this looks like gratuitous complication.

Reply:
Haha thanks for spotting this, my intention was to avoid create many Datum instances, but I forgot to check the count of the key. I'll fix this.